packages/core/test/invoke.test.ts

```diff
@@ -1185,6 +1224,10 @@ describe('invoke', () => {
                           src: 'getRandomNumber',
                           onDone: {
                             target: 'success',
+                            // TODO: investigate why event is DoneInvoke<any> instead of
+                            // DoneInvoke<{result: ...}>
+                            // The type is properly inferred for:
+                            // actions: ({ event }) => event.output.result
```

Comment:
nested inference contexts strike again 😬 

I think the conditional type that distributes over "non-fixed" (aka not yet set in stone) `TActors` is at fault here. I prepared a somewhat minimal repro of this problem: [TS playground](https://www.typescriptlang.org/play?ts=5.1.3#code/JYOwLgpgTgZghgYwgAgIILAeygSQLYAOANsgN4BQyyAzlAgFw1hSgDmA3JcpgK5gF9GcEAE9OVAPQSq3PgLCMeIANYhMAdxCcAvuXKhIsRCgAimEBBwgAbpmUQAotYjgAPABUTcMHAB8ZLjARAghGAAMAE3MIADpQW3sYgBJSamY2bTDxWX5BZE9vOB09IJC0ampgVhB0MGBzAHkAIwArCAwPJxcwZAgAD0gQCOoyZFLQphYQVmRtfwBeAKpxxgAiPrTvWLgKqpBV7IhrRncu8GyACiOTs7AASkZbYAidTnJxtAx6kAAxJS-zJ1nOBegMXMNRitJmxZgtkFdjvlbndkPN-E8XnoIu0iHAoCgYP86uZkDtKtUgd1QYMIaQxsEJmkpjM5hcuGS9tBGAibsD7qj0ZhnuQHuVyTUASBmm0Oqc+b43lJkEqohBqCAAOQ9dTYZTIdTAMAACzGRuAI2seOAwh6EENRug5BVOLxBKJ31Juwpcqp-RpIzpUKZMNZTukVA51S58OuSL5KLRyAxYeQotq3z+IEllPACpKDM+xJAOdtYKGAfpIUYwemsNRXAAPmK9unGq12mAS75G4WM+7AT7c29sQhcfjkISs0XkAh8VsALKIM0WDy1bAjP3gkZr3CEIi+C4IcwwKqMCjLBnUM9cCMYdcnHfUbLabKbSBX5AAJXa2Airhv0LTAANAB54yFQ8R2BM7iPtSW6jLQDDIKAMDQPkADKdCzAB4EAPxLOBhFUIhJyYQg2REeB5hmBYjCtsWNGWDYUG3Kud5QNQADaqy8LkYCrAAur4eaUTIuiiYwFjOFAFHYVQebaKKGJvLOEALkuoAQBcYHjB+YG3lgHFntonoEeBJHIKsrAQGAn7CFEeAAHI8HgTTQAcOE5PIZ7IPi1A8EQCjICALluVAsyydoIFUFFXBvmq17gXAiWEZB9gpURFlWTZdlDJgTmhe50WUdR0Tckc3QJv4+mURV5yeeBSoAHq4Q1szFeBsWdcVsWKSpc6QIuCDLlpOmXhlpLsXpJk7GZMhZdZtn2flzmue5slULx3mjH5AVBSFa3hS+AFde1cU+O+E3JXNEHMelN3zXQayLblDmrWFqwdYRpW0Z64oXAilUCg94F1WAG2Ec1rWiYpX0xV9p29XcnBAA)

Note that the type for that action object is pulled directly from the constraint, if we do this:
```diff
interface ActorImpl {
  src: string;
-  output: any;
+  output: any;
}
```
then we'll see `DoneInvokeEvent<unknown>` instead of `DoneInvokeEvent<any>`.

Reply:
Is there any workaround to this problem?

Reply:
Convince TS team to merge my https://github.com/microsoft/TypeScript/pull/48838 😆 see how it works just alright in the playground that uses that PR: [TS playground](https://www.staging-typescript.org/play?lib=lib.esnext.d.ts&ts=5.2.0-pr-48838-5#code/JYOwLgpgTgZghgYwgAgIILAeygSQLYAOANsgN4BQyyAzlAgFw1hSgDmA3JcpgK5gF9GcEAE9OAX3LlQkWIhQARTCAg4QAN0wBrCAFF1EcAB4AKgrhg4APjJcwIghEYADACbKIAOlCadngCSk1Mxs4s6cVLz8gshmFnASUvaOaNTUwKwg6GDAygDyAEYAVhAYpvqGYMgQAB6QIK7UZMjJTkwsIKzI4jYAvLZUrYwARDXBFl5waRkgwxHV6owmFeDzABQQi7ErYACUjJrArhKc5K1oGLkgAGI8IJfK5Qbg1XWGjc1D7WzdfcgbW2Wzz2yF6NkOxykrlKRDgUBQMDuDxAyCm6UyT0qr3qH1ILQcbWCHS6PTWXDRM2gjABSx2u1B4MwR3I+1S6KyyMKJTKQMqVlO52yV0xL1qOKaeK+RJ+PVBXAAPmyZkL8sVSmARWArAqLjllLd7nqQJr+VCYXCEUijcgEPCJgBZRAAC1AEFM2WwTTF7yaHtwhCIVjWCGUMAyjAogwJ1AjXCoiCwUBjsT91Hm4nm40gyYASqVsK4jHHvp0ADTFyNUKs+bRtEyp7E+5q0BjIUAwaCxADKdG6xaryAA-AMB6OqC2lj2EPMxwPlEoVIwVcaF6oNLWdu6MJ6ANrDKICMDDAC6VlNs6rkgvjBUBigM77VFN4lZENOtogDudrrWldaycreNtyTCNxFRCV+xoOgRlYCAwBzYR3DwAA5Hg8AKaA5kgg8YjxeFqB4IgwBvNCMKgboH3EcsqCorgswgADizgWMxxrHQWNnCdkGGWD4MQzAUNIzDqNnecPGpTZKnpMERwvSTVkggcAHolOQAA9QdFO6ESB1o3SRNol93ztSBHQQF0VF-Oxow4oDEwAsCplkqsuJ4uCEIaATUPQzCH0iPhDwjZB8MI4jkBAITyIzYs9O0ujLGzWzUSSqg2LaQDR1c3iPKQ7yyOGHTRzExdwPZNYASkhlnLHeSwD80cVPUzSLxfQqaMK2LDN2TggA)

On a serious note, I'd have to spend some time to figure this out for the current TS capabilities (if it's even possible)

Reply:
This one still falls back to the constraint - despite the fact that they have callable signatures already. I doubt that this is something that we can control rn without landing the mentioned fix and all I can do is to ping the TS team periodically to give it another look 🤷‍♂️

Reply:
That PR of mine mentions that the position of the conditional type can change the inferred result - so I guess I still have to try this out.

Reply:
I tried moving this conditional type in a variety of ways and couldn't make it work. I think it's best to assume, for now, that it's not possible to improve this and we can revisit this later.